### ISSUE TITLE
Breaking changes in version 4?

### ISSUE THREAD

**Issue by hugihlynsson** [intercom/intercom-node#342]
Hi! We just noticed that you've released version 4 of `intercom-node` but we're having some issues identifying wether we are able to upgrade without modifications or not. 

The major version updates hits at this being a breaking update but the [change-log entry for 4.0](https://github.com/intercom/intercom-node/releases/tag/v4.0.0) doesn't really help us understand what parts we should need to update. The [migration doc](https://github.com/intercom/intercom-node/wiki/Migration-guide) only mentions version 3. 

Can you help us understand what changes we might have to make to upgrade from version 3 to 4?

**Comment by colmdoyle** [comment]
Great callout @hugihlynsson, and definitely on us to fix for you.

tldr - Yes, `4.0.0` is definitely a breaking change, the biggest ones being that as of [API v2.6](https://developers.intercom.com/intercom-api-reference/v2.6/reference/changelog#improved-conversation-list-pagination) `client.conversations.list` now uses a cursor based pagination system instead of just page numbers.

Also we spotted a typo in [one of our](https://github.com/intercom/intercom-node/commit/2fb68b85501748c66155cafb0283c2dc31ad02cd) `enum` value, so if you rely on that you'll start seeing TS errors

**Comment by SeanHealy33** [comment]
Hi @hugihlynsson to go a bit deeper on the specific changes to conversations

**OLD**
```
const response = await client.conversations.list({
    order: Order.DESC,
    sort: SortBy.UpdatedAt,
    page: 1,
    perPage: 10,
});
```

**NEW**
```
const response = await client.conversations.list({
    startingAfter: 'WzE2NzA0MjI1MjkwMDAsMjQzMTY3NzA2ODcsMl0=',
    perPage: 10,
});
```

The new list uses a `startingAfter` param to get the next page instead of the old `page` parameter (the next starting_after key is provided on each page as you get it). As of `4.0` we also removed `page` `order` and `sort` from the list method here since they no longer do anything as of API Version 2.6

**Comment by hugihlynsson** [comment]
That's the info we needed. Thanks!
